python/paddle/distribution/gumbel.py

```diff
@@ -0,0 +1,242 @@
+# Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import numbers
+import math
+import numpy as np
+
+from paddle.distribution.transformed_distribution import TransformedDistribution
+from paddle.fluid import framework as framework
+
+
+class Gumbel(TransformedDistribution):
+    r"""The Gumbel distribution with location `loc` and `scale` parameters.
+
+    Mathematical details
+
+    The probability density function (pdf) is
+
+    .. math::
+
+        pdf(x; mu, sigma) = exp(-(x - mu) / sigma - exp(-(x - mu) / sigma)) / sigma
+
+
+    In the above equation:
+
+    * :math:`loc = \mu`: is the mean.
+    * :math:`scale = \sigma`: is the std.
+
+    Args:
+        loc(int|float|tensor): The mean of gumbel distribution.The data type is int, float, tensor.
+        scale(int|float|tensor): The std of gumbel distribution.The data type is int, float, tensor.
+
+    Examples:
+        .. code-block:: python
+
+          import paddle
+          from paddle.distribution.gumbel import Gumbel
+
+          # Gumbel distributed with loc=0, scale=1
+          dist = Gumbel(paddle.full([0.0]), paddle.full([1.0]))
```

Comment:
`paddle.full([0.0])` will report error? use `paddle.full([1], 0.0)` instead?

Reply:
`paddle.full([1.0])` is also needed to correct.

Reply:
done.